ldes-server-port-fetch/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fetching/services/CompactionCandidateComparator.java

```diff
@@ -0,0 +1,60 @@
+package be.vlaanderen.informatievlaanderen.ldes.server.fetching.services;
+
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.model.LdesFragmentIdentifier;
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.model.TreeRelation;
+import be.vlaanderen.informatievlaanderen.ldes.server.fetching.entities.CompactionCandidate;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static be.vlaanderen.informatievlaanderen.ldes.server.fragmentation.services.FragmentComparator.hasNoConnections;
+
+public class CompactionCandidateComparator {
+	public static Stream<CompactionCandidate> sortCompactionCandidates(Stream<CompactionCandidate> compactionCandidates) {
+		List<CompactionCandidate> candidatesList = compactionCandidates.toList();
+
+		var firstElement = candidatesList.stream()
+				.filter(candidate -> hasNoConnections(candidate.getFragment(), candidatesList.stream()
+						.map(CompactionCandidate::getFragment)
+						.toList()))
+				.findFirst()
+				.orElseThrow();
+
+		var map = candidatesList.stream()
+				.filter(candidate -> !candidate.getFragment().getRelations().isEmpty())
+				.collect(Collectors.toMap(CompactionCandidate::getId, candidate -> candidate.getFragment().getRelations()
+						.stream()
+						.findFirst()
+						.map(TreeRelation::treeNode)
+						.map(LdesFragmentIdentifier::asString)
+						.orElseThrow()));
```

Comment:
We should not throw exceptions in a background job as they will just go unnoticed. This is an issue in a lot of places in the server at the moment. Maybe we can have some clear error logging + a metric instead? Then your grafana dashboard will tell you IF and WHEN issues occurred and the logs will tell WHAT happened.